tests/http_base.py

```diff
@@ -20,7 +20,16 @@ class DaphneTestCase(unittest.TestCase):
     ### Plain HTTP helpers
 
     def run_daphne_http(
-        self, method, path, params, body, responses, headers=None, timeout=1, xff=False
+        self,
+        method,
+        path,
+        params,
+        body,
+        responses,
+        headers=None,
+        timeout=1,
+        xff=False,
+        raw_path=False,
```

Comment:
I'm thinking we don't need to add the `raw_path` parameter here. If you look at [the base implementation in Twisted](https://github.com/twisted/twisted/blob/4bceaac10570db717ee34a7cf63a5220881ab728/src/twisted/web/http.py#L871-L876), `self.path` that gets unquoted and added to the scope _just is_ what you want for `raw_path`. 

So, a test checking that `self.path` gets correctly added to the scope. (And the addition to `required_keys` in `assert_valid_http_scope()`) Probably looks enough. 

(The tests with the likes of `"/te st-à/"` are a bit misleading I think, since Daphne would never receive that, AFAICS right now.)

Reply:
Ah, but the testing harness doesn't build the scope the same way... (Sigh. 🙂)